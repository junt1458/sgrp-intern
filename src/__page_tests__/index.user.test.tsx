import IndexPage from '../pages';
import { render } from '@testing-library/react';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      push: jest.fn(),
    };
  },
}));

jest.mock('@auth0/auth0-react', () => ({
  Auth0Provider: ({ children }) => <div>{children}</div>,
  useAuth0: () => ({
    isLoading: false,
    user: { sub: 'foobar' },
    isAuthenticated: true,
    loginWithRedirect: jest.fn(),
    getAccessTokenSilently: jest.fn().mockReturnValue({ then: jest.fn() }),
  }),
}));

it('should render the component', () => {
  const { container } = render(<IndexPage />);
  expect(container).toMatchSnapshot();
});
