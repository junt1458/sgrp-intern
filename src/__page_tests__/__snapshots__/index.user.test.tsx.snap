// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component 1`] = `
<div>
  <div
    class="flex h-14 w-full items-center justify-between bg-slate-900 px-4 py-2 text-xl"
  >
    Internship Application System
    <button
      title="Logout"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LogoutIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
        />
      </svg>
    </button>
  </div>
  <div
    class="my-8"
  >
    <div
      class="flex flex-wrap items-center justify-center"
    >
      <img
        alt="Loading icon"
        data-nimg="1"
        decoding="async"
        height="200"
        loading="lazy"
        src="/_next/image?url=%2Floading.gif&w=640&q=75"
        srcset="/_next/image?url=%2Floading.gif&w=256&q=75 1x, /_next/image?url=%2Floading.gif&w=640&q=75 2x"
        style="color: transparent;"
        width="200"
      />
      <div
        class="w-full text-center"
      >
        Redirecting to home...
      </div>
    </div>
  </div>
</div>
`;
